### PR TITLE
Random rejections

### PR DESCRIPTION
**Pull request by danieldeidda**
Moving the random rejection utility for list mode from the STIR experimental

### CODE REVIEW

File: src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include "stir_experimental/listmode/LmToProjDataWithRandomRejection.h"
+
+USING_NAMESPACE_STIR
+
+
+
+int main(int argc, char * argv[])
+{
+  
+  if (argc<1 && argc>3) {
```
Comment (KrisThielemans) [inline src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx]:
```suggestion
  if ((argc<1) || (argc>3)) {
```

File: src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//
+/*!
+  \file 
+  \ingroup listmode
+
+  \brief Program to bin listmode data to projection data using bootstrapping (uses stir::LmToProjDataWithRandomRejection)
+ 
+  \author Kris Thielemans
+  
+  $Revision $
```
Comment (KrisThielemans) [inline src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx]:
```suggestion
```

File: src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+//
+//
+/*!
+  \file 
+  \ingroup listmode
+
+  \brief Program to bin listmode data to projection data using bootstrapping (uses stir::LmToProjDataWithRandomRejection)
```
Comment (KrisThielemans) [inline src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx]:
```suggestion
  \brief Program to bin listmode data to projection data using random rejection of counts (uses stir::LmToProjDataWithRandomRejection)
```

File: src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+USING_NAMESPACE_STIR
+
+
+
+int main(int argc, char * argv[])
+{
+  
+  if (argc<1 && argc>3) {
+    std::cerr << "Usage: " << argv[0] << " [par_file reject_if_above]]\n";
```
Comment (KrisThielemans) [inline src/listmode_utilities/lm_to_projdata_with_random_rejection.cxx]:
```suggestion
    std::cerr << "Usage: " << argv[0] << " [par_file fraction_of_counts_to_keep]]\n";
```

**Comment by KrisThielemans** [comment]
github won't let me commit my own suggestions... oh well. can you please batch them up and commit?

Note: we'll squash-merge this, as you still have weird commits in your history.

**Comment by KrisThielemans** [comment]
hmmm. did you just copy the files? Seems the original ones are still in `experimental`. should have used `git mv`. I guess it would be fine to now `git rm` (and remove the files from the `CMakeLists.txt` in `experimental`), i.e. I believe it doesn't really matter if we moved by git or 2 stage copy/delete when doing a squash-merge. @casperdcl is that correct?

**Comment by danieldeidda** [comment]
> 
> 
> hmmm. did you just copy the files? Seems the original ones are still in `experimental`. should have used `git mv`. I guess it would be fine to now `git rm` (and remove the files from the `CMakeLists.txt` in `experimental`), i.e. I believe it doesn't really matter if we moved by git or 2 stage copy/delete when doing a squash-merge. @casperdcl is that correct?

Yes I actually copied those files. If i am not mistaken that utility was actually commented on experimental

**Comment by casperdcl** [comment]
@KrisThielemans yes if you squash then it doesn't matter. Mind you I'd prefer rebasing the two commits into one because history is nice. Actually because of compression even keeping the commits separate as-is won't increase the clone size.

**Comment by KrisThielemans** [comment]
@casperdcl thanks for the help. I very much agree with  keeping history. However, for some reason @danieldeidda has 2 commits in his PR (a 2017 one and a later one that reverts it), which I'd have to get rid of. Also, here history is really only 2 commits: move files with clean-up, add doc. I'd have to reorder, and squash some. Sadly, life is too short for me to (remember how to) interactive rebase...

**Pull request merged**